hooks/dhclient-stop.in

```diff
@@ -0,0 +1,15 @@
+#! /bin/bash
+
+set -e
+
+LXC_HOOK_DIR="@LXCHOOKDIR@"
+
+rootfs="${LXC_ROOTFS_PATH##*:}"
+pidfile="${rootfs%/*}/dhclient.pid"
+
+# XXX Stop hook namespace arguments are wrong for some reason, those are the host namespaces not the container ones.
```

Comment:
Hi,  from the lxc.container.conf manpage:

1448           In the case of the stop hook, paths to filedescriptors
1449           for each of the container's namespaces along with their types
1450           are passed. </para></listitem>

This is because the init task is supposed to be destroyed at this point, so you can't use the initpid, obviously.

But you could use the passed-in ns fd arguments, at least that's the point of them.

But, if you have the dhclient process sitting around to inspect, that works too...  What do you think, keep as is or switch to using the fds?

Reply:
Sorry that's what I meant in my comment, I tried to use the fds at first but they were wrong for some reason and I didn't have time to look into it.

Reply:
Note that before the namespaces are passed as arguments there are additional arguments which @hallyn intended to remove but didn't yet afaik. (We should do that soon btw.). So the hook gets passed:

    <container-name> lxc stop <path-to-ns-1> <path-to-ns-1> ...

Actually it probably would have been smarter to just set these as:

    LXC_MOUNT_NS=<path-to-fd>
    LXC_CGROUP_NS=<path-to-fd>
    LXC_USER_NS=<path-to-fd>
    .
    .
    .

Reply:
All right since the lxc package will ship the hooks, we can fix those up when we fix what gets sent to the hooks.

(I agree sending as LXC_MOUNT_NS etc in env is better)